packages/upload/test/keyboard-navigation.common.js

```diff
@@ -31,61 +31,108 @@ describe('keyboard navigation', () => {
     uploadElement.files = [FAKE_FILE];
 
     await nextRender();
-
-    fileElement = uploadElement.querySelector('vaadin-upload-file');
+    uploadButton = document.querySelector('vaadin-button[slot=add-button]');
+    fileElements = uploadElement.querySelector('vaadin-upload-file');
   });
 
   afterEach(() => {
     button.focus();
   });
 
   it('should focus on the upload button', async () => {
-    const uploadButton = uploadElement.shadowRoot.querySelector('[part=upload-button]');
-
     await repeatTab(1);
 
-    expect(uploadElement.shadowRoot.activeElement).to.equal(uploadButton);
+    expect(document.activeElement).to.not.equal(null);
+    expect(document.activeElement).to.equal(uploadButton);
```

Comment:
There is no need to check for `null`, the second check is enough.

Reply:
UPD: the previous test was incorrect due to `upload-button` part being removed in #4857.
So we discussed that the explicit `null` check might be OK to keep for now.